compiler/resolution/functionResolution.cpp

```diff
@@ -6246,6 +6329,11 @@ static void resolveNewHandleGenericClass(CallExpr* newExpr) {
 
   resolveNewGenericInit(initCall, at, initTemp);
 
+  if (FnSymbol* fn = initCall->resolvedFunction())
+    if (fn->hasFlag(FLAG_PROMOTION_WRAPPER))
+      USR_FATAL(initCall,
+                "Promoted initialization of generic classes not implemented");
```

Comment:
Can we add a .future or something for this error message?

Reply:
Oh sorry I meant to remove that statement - the check doesn't work. This PR adds futures 
    test/classes/initializers/promotion/generic-promote-new.chpl
    test/classes/initializers/promotion/concrete-promote-new.chpl
 but it's mainly left as future work in issue #9772.